bbq/vm/vm.go

```diff
@@ -695,20 +696,57 @@ func opNew(vm *VM, ins opcode.InstructionNew) {
 	// TODO: Support inclusive-range type
 	compositeStaticType := staticType.(*interpreter.CompositeStaticType)
 
+	compositeFields := newCompositeValueFields(vm, compositeKind)
+
 	value := interpreter.NewCompositeValue(
 		vm.config,
 		EmptyLocationRange,
 		compositeStaticType.Location,
 		compositeStaticType.QualifiedIdentifier,
 		compositeKind,
-		nil,
+		compositeFields,
 		// Newly created values are always on stack.
 		// Need to 'Transfer' if needed to be stored in an account.
 		common.ZeroAddress,
 	)
 	vm.push(value)
 }
 
+func newCompositeValueFields(vm *VM, compositeKind common.CompositeKind) (fields []interpreter.CompositeField) {
```

Comment:
Maybe only pass the `UUIDHandler` instead of the entire vm

Reply:
Just a nit: can we move this function to a separate file, say `vm/value_composite.go` and have all composite value related utilities in it (similar to `vm/value_account.go`, `vm/value_capability.go ). Then we can keep `vm.go` to only contain functions specific to the vm/vm-operations.

Reply:
Good idea moving this to a separate file and keeping `vm.go` small and focused. 👌 

The UUID handler is only needed for resources, and a future version of the function will very likely need other functionality when it will support other composite kinds. I've changed it to just pass the config for now.